flaxandteal/coral-arches
# Fix/2317 prevent functions from generating cardinality 1 tiles

**Description**

Depends on: https://github.com/flaxandteal/arches/pull/14

Utilizes the new escape_function parameter that can be used when importing business data into the application. The pattern I used to identify if a function would be a problem is:

- Does it trigger on a nodegroup
- Does it create a tile with a different nodegroup to the trigger which is cardinality 1
- The function will need escaped

**Tests**
- Create a new monument using the add monument workflow
- This will have created a new display name and ha reference tile (both cardinality 1 tiles)
- Also create an SMR, Garden Number and HB Number
- Now you should have multiple tiles that are all configured with functions creating cardinality 1 tiles
- Export your Monument
- Attempt to import using the escape function flag
- `make manage CMD="packages -o import_business_data -s data.json -ow overwrite --escape_function"`
- Everything should work correctly first time
- You should be able to run the import command multiple times without needing to clear your instances
